crypto/crypto/src/hash.rs

```diff
@@ -248,6 +248,7 @@ impl HashValue {
 
     /// Returns the `index`-th nibble.
     pub fn get_nibble(&self, index: usize) -> Nibble {
+        precondition!(index < HashValue::LENGTH);
```

Comment:
Can you put this one in a separate (first) commit, please? That way we can manage tag analysis and bound checks independently (mentally, in history, in reverts, ...).